core/domain/user_jobs_one_off.py

```diff
@@ -135,6 +139,97 @@ def reduce(key, stringified_values):
                 subscription_services.subscribe_to_collection(key, item['id'])
 
 
+class UserImpactCalculationOneOffJob(jobs.BaseMapReduceJobManager):
+    """ Calculates each user's impact score, where impact score is defined as:
+    Sum of (
+    ln(playthroughs) * (ratings_scaler) * (average(ratings) - 2.5))
+    *(multiplier),
+    where multiplier = 10, and ratings_scaler is .1(number of ratings)
+    if there are < 10 ratings for that exploration.
+
+    Impact scores are calculated over explorations for which a user
+    is listed as a contributor.
+    """
+
+    MULTIPLIER = 10
+    MIN_AVERAGE_RATING = 2.5
+    NUM_RATINGS_SCALER_CUTOFF = 10
+    NUM_RATINGS_SCALER = .1
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @classmethod
+    def _get_exp_impact_score(cls, exploration_id):
+        from core.domain import rating_services
+        from core.domain import stats_jobs_continuous
+
+        # Get ratings and compute average rating score.
+        rating_frequencies = rating_services.get_overall_ratings_for_exploration(
+            exploration_id)
+        total_number = 0
+        total_rating = 0.0
+        for rating, num_ratings in rating_frequencies.iteritems():
+            total_rating += (int(rating) * num_ratings)
+            total_number += num_ratings
+        # Only divide by a non-zero number.
+        if total_number == 0:
+            return 0
+        average_rating = total_rating / total_number
+
+        # Get rating term to use in impact calculation.
+        rating_term = average_rating - cls.MIN_AVERAGE_RATING
+        # Only explorations with an average rating greater than the minimum
+        # have an impact.
+        if rating_term <= 0:
+            return 0
+
+        # Get num_ratings_scaler.
+        if total_number < cls.NUM_RATINGS_SCALER_CUTOFF:
+            num_ratings_scaler = cls.NUM_RATINGS_SCALER*(total_number)
+        else:
+            num_ratings_scaler = 1
+
+        # Get number of completions/playthroughs.
+        num_completions = stats_jobs_continuous.StatisticsAggregator.get_statistics(
+            exploration_id, 'all')['complete_exploration_count']
```

Comment:
Hmmm. Can you make stats_jobs_continuous._VERSION_ALL public (remove the leading underscore) and refer to that instead?